src/basho_bench.erl

```diff
@@ -93,11 +91,35 @@ run_benchmark(Configs) ->
 
     log_dimensions(),
 
+    ok.
+
+
+run_benchmark() ->
     basho_bench_sup:start_child(),
-    ok = basho_bench_stats:run(),
-    ok = basho_bench_measurement:run(),
-    ok = basho_bench_duration:run(),
-    ok = basho_bench_worker:run(basho_bench_worker_sup:workers()),
+    case {master_node(), node_count()} of
+        %% No master
+        {undefined, _} ->
+            bootstrap_bb(),
+            ok = run_benchmarks(basho_bench_worker_sup:workers());
+        %% Only one node, equivalent to no master
+        {_, 1} ->
+            bootstrap_bb(),
+            ok = run_benchmarks(basho_bench_worker_sup:workers());
+        %% Master and multiple workers
+        {MasterNode, NodeCount} when NodeCount > 1 ->
+            case MasterNode =:= node() of
+                true ->
+                    await_nodes(NodeCount),
+                    bootstrap_bb(),
+                    ok = run_benchmarks();
+                false ->
+                    ?INFO("Worker node[~p] connecting to master node: ~p~n", [node(), MasterNode]),
+                    pong = net_adm:ping(MasterNode),
+                    true = length(nodes()) > 0,
```

Comment:
This is basically an explicit assertion so we crash if we've gotten this far, are not the master, and there are no nodes joined to the bench cluster?

Reply:
Correct, it's somewhat redundant in that in the above line we're asserting that we're able to ping `MasterNode`, but it's nice to be explicit that we've got properly connected nodes.